proc/proc_windows.go

```diff
@@ -308,19 +313,23 @@ func (dbp *Process) parseDebugLineInfo(exe *pe.File, wg *sync.WaitGroup) {
 	}
 }
 
+var UnsupportedArchErr = errors.New("unsupported architecture of windows-386 - only windows-amd64 is supported")
+
 func (dbp *Process) findExecutable(path string) (*pe.File, error) {
 	if path == "" {
 		// TODO: Find executable path from PID/handle on Windows:
 		// https://msdn.microsoft.com/en-us/library/aa366789(VS.85).aspx
 		return nil, fmt.Errorf("not yet implemented")
 	}
-	f, err := os.OpenFile(path, 0, os.ModePerm)
-	if err != nil {
-		return nil, err
+	peFile, err := openExecutablePath(path)
+	if err == nil {
+		dbp.dwarf, err = dwarfFromPE(peFile)
+		if err != nil {
+			return nil, err
+		}
 	}
-	peFile, err := pe.NewFile(f)
-	if err != nil {
-		return nil, err
+	if peFile.Machine != pe.IMAGE_FILE_MACHINE_AMD64 {
```

Comment:
`peFile` could be nil here. We must explicitly check for an error, or check that this pointer is not nil.